compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -1639,28 +1633,29 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
     }
 
     /** generates a sequence of JS constructor statements based on a constructor tree. */
-    private def genJSClassCtorStats(overloadVar: js.VarRef, primaryCtor: PrimaryJSCtor,
-        secondaryCtorTrees: List[ConstructorTree])(implicit pos: Position): js.Tree = {
+    private def genJSClassCtorStats(overloadVar: js.VarRef,
+        ctorTree: ConstructorTree[PrimaryJSCtor])(implicit pos: Position): js.Tree = {
 
       /* generates a statement that conditionally executes body iff the chosen
        * overload is any of the descendants of `tree` (including itself).
        *
        * here we use the property from building the trees, that a set of
        * descendants always has a range of overload numbers.
        */
-      def ifOverload(tree: ConstructorTree, body: js.Tree): js.Tree = body match {
+      def ifOverload(tree: ConstructorTree[_], body: js.Tree): js.Tree = body match {
         case js.Skip() => js.Skip()
 
         case body =>
-          val lo = tree.lowestNum
-          val hi = tree.overloadNum
-          assert(lo <= hi, "bad overload range")
-
           val x = overloadVar
           val cond = {
-            if (lo == hi) {
-              js.BinaryOp(js.BinaryOp.Int_==, js.IntLiteral(lo), x)
+            if (tree.subCtors.isEmpty) {
+              js.BinaryOp(js.BinaryOp.Int_==, js.IntLiteral(tree.overloadNum), x)
             } else {
+              val lo = tree.overloadNum
+              val hi = tree.subCtors.last.overloadNum
```

Comment:
I don't think this is correct. If the last child constructor has subconstructors of its own, its `overloadNum` will be smaller than its children's, but `hi` should be the highest value of them all.